### PR TITLE
Added param ignore_throttled=false when indicesOptions.ignoreThrottle…

### LINKED ISSUE #42358: High Level client dont include `ignore_throttled` as parameter
SearchRequest don't include ignore_throttled=false in the request.

` SearchRequest searchRequest = new SearchRequest();
        searchRequest.indicesOptions(IndicesOptions.fromOptions(true, true,
                true, false, searchRequest.indicesOptions().allowAliasesToMultipleIndices(),
                searchRequest.indicesOptions().forbidClosedIndices(), searchRequest.indicesOptions().ignoreAliases(), false));
    SearchResponse response = client.search(searchRequest, RequestOptions.DEFAULT);
 `

### PR DESCRIPTION
**Pull request by kfuksman**
closes #42358

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java [hunk 1]
```diff
@@ -939,6 +939,9 @@ Params withIndicesOptions(IndicesOptions indicesOptions) {
                     expandWildcards = joiner.toString();
                 }
                 putParam("expand_wildcards", expandWildcards);
+                if(!indicesOptions.ignoreThrottled()){
```
Comment (martijnvg) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: I think the `ignore_throttled` should always be serialized in the context here:

```
putParam("ignore_throttled", Boolean.toString(indicesOptions.ignoreThrottled()));
```

Whether `ignore_throttled` is enabled or disabled by default depends on the api, and the `withIndicesOptions (...)` method is used for multiple APIs.

**Review by martijnvg — COMMENTED** [review]
Hey @kfuksman Thanks for open a pr this fix this! I left a comment on this PR.

Also the `RequestConvertersTests` should updated to test this change properly.
Specifically the `setRandomIndicesOptions(...)` methods should randomly set the `ignore_throttled` parameter.

```patch
Index: client/rest-high-level/src/test/java/org/elasticsearch/client/RequestConvertersTests.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- client/rest-high-level/src/test/java/org/elasticsearch/client/RequestConvertersTests.java	(date 1558690291000)
+++ client/rest-high-level/src/test/java/org/elasticsearch/client/RequestConvertersTests.java	(date 1558697501000)
@@ -1561,7 +1561,7 @@
         endpoint.add("_field_caps");
 
         assertEquals(endpoint.toString(), request.getEndpoint());
-        assertEquals(4, request.getParameters().size());
+        assertEquals(5, request.getParameters().size());
 
         // Note that we don't check the field param value explicitly, as field names are
         // passed through
@@ -1595,7 +1595,7 @@
         }
         endpoint.add(RestRankEvalAction.ENDPOINT);
         assertEquals(endpoint.toString(), request.getEndpoint());
-        assertEquals(3, request.getParameters().size());
+        assertEquals(4, request.getParameters().size());
         assertEquals(expectedParams, request.getParameters());
         assertToXContentBody(spec, request.getEntity());
     }
@@ -1922,7 +1922,8 @@
                                         Map<String, String> expectedParams) {
 
         if (randomBoolean()) {
-            setter.accept(IndicesOptions.fromOptions(randomBoolean(), randomBoolean(), randomBoolean(), randomBoolean()));
+            setter.accept(IndicesOptions.fromOptions(randomBoolean(), randomBoolean(), randomBoolean(), randomBoolean(),
+                true, false, false, randomBoolean()));
         }
         expectedParams.put("ignore_unavailable", Boolean.toString(getter.get().ignoreUnavailable()));
         expectedParams.put("allow_no_indices", Boolean.toString(getter.get().allowNoIndices()));
@@ -1935,11 +1936,13 @@
         } else {
             expectedParams.put("expand_wildcards", "none");
         }
+        expectedParams.put("ignore_throttled", Boolean.toString(getter.get().ignoreThrottled()));
     }
 
     static IndicesOptions setRandomIndicesOptions(IndicesOptions indicesOptions, Map<String, String> expectedParams) {
         if (randomBoolean()) {
-            indicesOptions = IndicesOptions.fromOptions(randomBoolean(), randomBoolean(), randomBoolean(), randomBoolean());
+            indicesOptions = IndicesOptions.fromOptions(randomBoolean(), randomBoolean(), randomBoolean(), randomBoolean(),
+                true, false, false, randomBoolean());
         }
         expectedParams.put("ignore_unavailable", Boolean.toString(indicesOptions.ignoreUnavailable()));
         expectedParams.put("allow_no_indices", Boolean.toString(indicesOptions.allowNoIndices()));
@@ -1952,6 +1955,7 @@
         } else {
             expectedParams.put("expand_wildcards", "none");
         }
+        expectedParams.put("ignore_throttled", Boolean.toString(indicesOptions.ignoreThrottled()));
         return indicesOptions;
     }
 
```

File: client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java (hunk 1 above)
Comment (kfuksman) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: Yes I agree with your changes. I need to do something more ? Sorry I’m newbie collaborating in open source

Comment (kfuksman) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java]: Thanks for your help! I agree with your changes and I already pushed these changes

**Comment by martijnvg** [comment]
Thanks @kfuksman! This looks great.

**Comment by martijnvg** [comment]
@elasticmachine Test this please

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/default-distro

**Comment by martijnvg** [comment]
@elasticmachine Test this please

**Comment by kfuksman** [comment]
@martijnvg it’s possible to tag to 6.7.x ? Or 6.8.x?

**Comment by martijnvg** [comment]
@kfuksman Yes, I will try to backport it to 6.8, 7.1 and 7.2 branches too.

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by martijnvg** [comment]
@elasticmachine Test this please

**Comment by martijnvg** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by martijnvg** [comment]
@elasticmachine Test this please

**Comment by kfuksman** [comment]
@martijnvg I need this change in 6.7.X or 6.8.x, what can I do? if I do this pr to branch 6.7 will be merged?

**Comment by martijnvg** [comment]
@kfuksman I'm backporting it to the 6.8 branch.

**Comment by martijnvg** [comment]
@kfuksman It has been backported to the 6.8 branch.

**Comment by kfuksman** [comment]
@martijnvg Thanks!

**Pull request merged**